charts/castai-agent/templates/clustervpa-deployment.yaml

```diff
@@ -0,0 +1,63 @@
+{{- if .Values.clusterVPA.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: castai-agent-cvpa
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: castai-agent-cvpa
+    app.kubernetes.io/instance: castai-agent-cvpa
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: castai-agent-cvpa
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: castai-agent-cvpa
+        app.kubernetes.io/instance: castai-agent-cvpa
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      {{- if .Values.priorityClass.enabled }}
+      priorityClassName: {{ .Values.priorityClass.name }}
```

Comment:
Thought: do we need priority for the autoscaler? Would it make sense to set it to a different value than agent?

Reply:
To keep backwards-compatible behavior, didn't see any harm to align priority/node selector with agent (especially node selector). Scaling agent is vital for it's uptime, so scaler should get at least equal priority imo.